Add Edge API page to Edge docs (#2629)
qdrant/landing_page: qdrant-landing/content/documentation/edge/edge-api/configuration.md

## EdgeConfig

```python
EdgeConfig(
    vectors: Optional[Union[EdgeVectorParams, Dict[str, EdgeVectorParams]]] = None,
    sparse_vectors: Optional[Dict[str, EdgeSparseVectorParams]] = None,
    on_disk_payload: Optional[bool] = None,
    hnsw_config: Optional[HnswIndexConfig] = None,
    quantization_config: Optional[QuantizationConfigType] = None,
    optimizers: Optional[EdgeOptimizersConfig] = None,
    max_search_threads: Optional[int] = None,
    search_pool_core: Optional[int] = None,
)
```

In Rust, `EdgeConfig` is a struct whose fields you can set directly, or build with the fluent `EdgeConfig::builder()`:

```rust
let config = EdgeConfig::builder()
    .vector("text", EdgeVectorParams::builder(384, Distance::Cosine).build())
    .on_disk_payload(true)
    .build();
```

| Parameter | Description |
|---|---|
| `vectors` | Dense vector configuration. In Python, a single `EdgeVectorParams` configures the default unnamed vector. Optional if `sparse_vectors` is given. |
| `sparse_vectors` | Sparse vector configuration. |

> | `on_disk_payload` | Store payload on disk with mmap rather than in RAM. Defaults to on-disk. |

Reviewer: ```suggestion
| `on_disk_payload` | Whether to cache payloads in RAM for faster access, or serve them from disk. |
```

| `wal_options` | A `WalOptions` value carrying the write-ahead log parameters. Rust only. Refer to [WAL Options](#wal-options). |

A new shard must define at least one of `vectors` or `sparse_vectors`; both are validated against the existing segments on `load`. Python raises `ValueError` if both are empty, so changing only a tunable parameter still requires redeclaring the vectors. Rust accepts a tunables-only configuration and takes the vectors from the shard.

## Dense Vector Parameters

`EdgeVectorParams` configures one named dense vector. `size` and `distance` are required and cannot be changed after the shard is created.

```python
EdgeVectorParams(
    size: int,
    distance: Distance,
    on_disk: Optional[bool] = None,
    multivector_config: Optional[MultiVectorConfig] = None,
    datatype: Optional[VectorStorageDatatype] = None,
    quantization_config: Optional[QuantizationConfigType] = None,
    hnsw_config: Optional[HnswIndexConfig] = None,
)
```

```rust
pub fn builder(size: usize, distance: Distance) -> EdgeVectorParamsBuilder
```

| Parameter | Description |
|---|---|
| `size` | Vector dimension. Required. |
| `distance` | Distance metric. Required. |

> | `on_disk` | Store vectors on disk with mmap rather than in RAM. |

Reviewer: ```suggestion
| `on_disk` | Whether to cache vectors in RAM for faster access, or serve them from disk. |
```
Vectors are always stored on disk. We should avoid "store" describing these "on disk" parameters.

| `multivector_config` | Multi-vector configuration, for late-interaction models. |
| `datatype` | Storage datatype for the vector. |
| `quantization_config` | Per-vector quantization, overriding the global setting. |
| `hnsw_config` | Per-vector HNSW parameters, overriding the global setting. |

## Sparse Vector Parameters

`EdgeSparseVectorParams` configures one named sparse vector. All parameters are optional.

```python
EdgeSparseVectorParams(
    full_scan_threshold: Optional[int] = None,
    on_disk: Optional[bool] = None,
    modifier: Optional[Modifier] = None,
    datatype: Optional[VectorStorageDatatype] = None,
)
```

```rust
pub fn builder() -> EdgeSparseVectorParamsBuilder
```

| Parameter | Description |
|---|---|
| `full_scan_threshold` | Threshold below which a full scan is used instead of the sparse index. |

> | `on_disk` | Store the sparse index on disk rather than in RAM. |

Reviewer: ```suggestion
| `on_disk` | Whether to cache sparse vector indexes in RAM for faster access, or serve them from disk.  |
```

    max_segment_size: Optional[int] = None,
    indexing_threshold: Optional[int] = None,
    prevent_unoptimized: Optional[bool] = None,
)
```

In Rust, set the fields you need and leave the rest at their defaults:

```rust
pub struct EdgeOptimizersConfig {
    pub deleted_threshold: Option<f64>,
    pub vacuum_min_vector_number: Option<usize>,
    pub default_segment_number: Option<usize>,
    pub max_segment_size: Option<usize>,
    pub indexing_threshold: Option<usize>,
    pub prevent_unoptimized: Option<bool>,
}
```

```rust
let optimizers = EdgeOptimizersConfig {
    indexing_threshold: Some(20_000),
    ..Default::default()
};
```

| Parameter | Description |
|---|---|
| `deleted_threshold` | Minimum fraction of deleted vectors in a segment required to run vacuum. Default: `0.2`. |
| `vacuum_min_vector_number` | Minimum number of vectors in a segment required to run vacuum. Default: `1000`. |
| `default_segment_number` | Target number of segments. `0` chooses automatically from the CPU count. |
| `max_segment_size` | Maximum segment size in KB. Derived from the CPU count when unset. |
| `indexing_threshold` | Size in KB above which a segment gets an HNSW index. |

> | `prevent_unoptimized` | Store points written to unoptimized segments larger than the indexing threshold as deferred points: they are persisted but excluded from reads and searches until the segments are optimized. |

Reviewer: ```suggestion
| `prevent_unoptimized` | [Prevents slow reads from large unoptimized segments](/documentation/ops-optimization/optimizer/#prevent-reads-from-large-unindexed-segments) by deferring the visibility of points until they've been indexed. |
```
Just a suggestion. I think a link here is helpful.
